Content/Skills/Magic/Fireball.cs

```diff
@@ -122,19 +263,482 @@ public override void OnKill(int timeLeft)
+				Vector2 position = Projectile.position + new Vector2(Main.rand.NextFloat(Projectile.width), Main.rand.NextFloat(Projectile.height));
+				Dust.NewDustPerfect(position, Main.rand.NextBool() ? DustID.IceTorch : DustID.Torch, vel, Scale: scale);
+			}
+		}
+
+		public override void ModifyHitNPC(NPC target, ref NPC.HitModifiers modifiers)
+		{
+			if (Owner.HasTreePassive<FireballTree, AbsoluteZero>() && (target.HasBuff<FreezeDebuff>() || target.HasBuff<IgnitedDebuff>()))
+			{
+				modifiers.FinalDamage += 0.35f;
+			}
+		}
+
+		public override void OnHitNPC(NPC target, NPC.HitInfo hit, int damageDone)
+		{
+			if (Owner.HasTreePassive<FireballTree, ThermalFeedback>())
+			{
+				Owner.statMana = Math.Min(Owner.statMana + 5, Owner.statManaMax2);
+			}
+		}
+
+		public override void OnKill(int timeLeft)
+		{
+			SpawnDust(2, 0.6f, true);
+
+			if (Owner.HasTreePassive<FireballTree, CrystallineImpact>())
+			{
+				int type = ModContent.ProjectileType<BlastIcicleSmall>();
+				int damage = Projectile.damage / 8;
+
+				for (int i = 0; i < 3; ++i)
+				{
+					Vector2 vel = Main.rand.NextVector2Circular(4, 4) - Projectile.velocity * 0.5f;
+					float scale = Owner.GetPassiveStrength<FireballTree, Rime>() * 0.3333f + 1;
+					int icicle = Projectile.NewProjectile(Projectile.GetSource_Death(), Projectile.Center, vel, type, damage, 0, Main.myPlayer, scale);
+					Main.projectile[icicle].GetGlobalProjectile<ElementalProjectile>().AddElementalValues((ElementType.Cold, 0, 1));
+				}
+			}
+
+			if (Owner.HasTreePassive<FireballTree, FrozenGround>())
+			{
+				Projectile.NewProjectile(Projectile.GetSource_Death(), Projectile.Center, Vector2.Zero, ModContent.ProjectileType<ColdAura>(), 0, 0, Projectile.owner);
+			}
+
+			int proj = ExplosionHitbox.QuickSpawn(Projectile.GetSource_Death(), Projectile, (int)(Projectile.damage * 0.75f), Projectile.owner, Projectile.Size * 2, ExplosionVfx);
+			ElementalProjectile ele = Main.projectile[proj].GetGlobalProjectile<ElementalProjectile>();
+			ele.AddElementalValues((ElementType.Fire, 0, 1), (ElementType.Cold, 0, 1));
+		}
+
+		public override bool PreDraw(ref Color lightColor)
+		{
+			Texture2D tex = TextureAssets.Projectile[Type].Value;
+			Vector2 position = Projectile.Center - Main.screenPosition;
+			Main.spriteBatch.Draw(tex, position, null, Projectile.GetAlpha(lightColor), Projectile.rotation, new Vector2(12, 30), 1f, SpriteEffects.None, 0);
+			return false;
+		}
+
+		public void PostHitNPC(NPC target, in NPC.HitInfo hit, int damageDone)
+		{
+			if (target.HasBuff<IgnitedDebuff>() && (target.HasBuff<FreezeDebuff>() || target.HasBuff(BuffID.Chilled)))
+			{
+				target.DelBuff(target.FindBuffIndex(ModContent.BuffType<IgnitedDebuff>()));
+
+				if (target.FindBuffIndex(ModContent.BuffType<FreezeDebuff>()) is not -1 and int freezeIndex)
+				{
+					target.DelBuff(freezeIndex);
+				}
+				else
+				{
+					target.DelBuff(target.FindBuffIndex(BuffID.Chilled));
+				}
```

Comment:
This seems to not consider that both Chilled and FreezeDebuff could theoretically be present. Also you can find indices right away instead of starting with Has checks.